algebraic_connectivity_notebook.ipynb

```diff
@@ -397,30 +402,53 @@
     "        out = model(data.x.type(torch.FloatTensor).to(device), data.edge_index, data.batch)\n",
     "        loss = criterion(out, data.y).item()  # Compute the loss.\n",
     "        #loss = check_accuracy_unsupervised(data, out, criterion).item()\n",
-    "        acc = check_accuracy(data,out)\n",
-    "    return loss, acc\n",
+    "        correct = 0\n",
+    "        start_slice = 0\n",
+    "        for i in range(len(data)):\n",
+    "            d = data[i]\n",
+    "            correct += check_accuracy(d, out[start_slice:start_slice+d.y.shape[0]])\n",
+    "            start_slice += d.y.shape[0]\n",
+    "    return loss, correct\n",
```

Comment:
Since we now have to handle both the whole dataset and a batch in the same way, this function is updated to take one graph at a time. `out` must be manually sliced using the number of nodes in `d` (we know that from the shape of target y).